ROADMAP.md

```diff
@@ -0,0 +1,226 @@
+# Docs WG Roadmap
+
+## Jan 2016 → Apr 2016 (Q1)
+
+This is the first of the quarterly roadmaps that this Working Group will
+produce, so it will be structured a bit differently than subsequent ones. Since
+this is a brand-new process, an explanation of **what it is** and **why it's
+necessary** follows.
+
+This roadmap outlines the priorities for the Docs working group for the
+quarter. These tasks can be as broadly or narrowly defined as is useful. They
+are not necessarily staffed — as folks step up to do the work, we'll link
+tracking issues to the tasks for them to self-organize. Every quarter we'll
+revisit this roadmap, adding a quick postmortem of how things went, and
+creating a new set of priorities for the ensuing quarter.
+
+That brings us to why a roadmap is necessary: the Node.js documentation is a
+relatively confined space to work within, and there are many interested
+parties: there are groups that consume the documentation as output, like the
+[website working group][wg-website], there are groups concerned with ensuring
+the [content addresses the needs of diverse audiences][wg-inclusivity], there are
+[groups that are required][wg-ctc] to produce API documentation as a
+pre-requisite to merging their primary work, there are contributors who wish to
+help in a _technical_ fashion, there are contributors who wish to contribute
+_editorially_, and there are the readers of the docs. All of these
+contributions overlap, so in order to make measurable progress without stepping
+on each other's feet, we have to:
+
+* **Set clear priorities** — if two contributions conflict, it should be
+  straightforward to determine which contribution should take precedence.
+* **Message these priorities well in advance** — with this many stakeholders,
+  we should mention repeatedly, loudly, and clearly what our intentions are.
+  The Docs WG should aspire to avoid surprise.
+* **Make the best use of donated time** — we want to ensure that if someone is
+  donating their time to improve the documentation, it will be well-spent, and
+  directed at driving the documentation towards this WG's [stated goals][goals].
+
+[Our goals][goals] can be summarized thusly:
+
+> Node.js should be in friendly competition for "Best Docs in OSS," with docs
+> that address the needs of a wide variety of audiences — across skill levels,
+> goals, and languages.
```

Comment:
No apologies necessary — it's totally my fault. Sorry for dropping the ball on getting that resolution out to where it needed to go. I'll make sure it doesn't happen again. I'm going to start an issue for the next meeting today, which was scheduled for Wednesday at 10AM PST next week. We can definitely revisit the timing of meetings along with that, since there are so many new members.

With regards to interactivity, it can be a double-edged sword. In the ES2015/ES5 switcher case, it's fine to use on the website's front page because the number of code examples there are bounded, as time goes on they can replace ES2015 with ES2016 and up, and due to the smaller number of examples, they can more easily curate that content. 

Applying the switcher to the docs is a big maintenance load: it can grow on the axis of number of language versions to support, it can grow on the number of code examples in the docs to be translated into different versions, and because it affects all code examples, it's represents a significant source of churn on the docs. 

The most limited resources the Docs WG have are **space** and **volunteered time**. Features that cause a lot of *churn* stand a high risk of blocking other work because the docs are a relatively constrained space. Maintaining separate examples — or being blocked by churn created by maintaining separate examples — spends volunteered time. Anything that takes a lot from both of those resources has to pay back that debt either by making it easier to work on the docs or by greatly increasing value for readers. Because the switcher represents a continual investment of author effort, it can't do the former, and because it doesn't offer any explanation of what the user is switching between, the value offered to readers is marginal — they would likely be better served by us redirecting them to a separate "learn javascript" document, either written by us, or maintained elsewhere (like jsforcats, or mdn.)

More generally, features that greatly increase the usefulness of the docs for readers while not continually spending author time are a net win. Some of the features proposed in the ROADMAP, like per-section YAML containing version information, represent a significant investment in churn and author time, but for the most part it's a one time cost. The ongoing cost can be mitigated by creating an automated linter rule to detect the absence of necessary information, which further reduces the time commitment. Meanwhile, version information is very valuable for users — they can now see at a glance when a feature was introduced, which lets them know immediately if it's available to them for use.